spec/rails_helper.rb

```diff
@@ -119,108 +97,131 @@ def stub_addresses
 
-  config.before(:each, type: :system) do
-    # Use truncation in the case of doing `browser` tests because it
-    # appears that transactions won't work since it really does
-    # depend on the database to have records.
-    DatabaseCleaner.strategy = :truncation
+    # Seed with default partner record
+    FactoryBot.create(:partner, organization: organization, name: default_records_data[:partners][:default][:name])
   end
 
-  config.before(:each, type: :request) do
-    # Use truncation in the case of doing `browser` tests because it
-    # appears that transactions won't work since it really does
-    # depend on the database to have records.
-    DatabaseCleaner.strategy = :truncation
-  end
+  # --------------------
+  # END - Seeding helpers for tests setup
+  # --------------------
 
-  config.before(:each) do |example|
-    if example.metadata[:persisted_data]
-      DatabaseCleaner.strategy = :truncation
-    else
-      # The database cleaner will now begin at this point
-      # up anything after this point when `.clean` is called.
-      DatabaseCleaner.start
-
-      # "Dirty" the database by adding the essential records
-      # necessary to run tests.
-      #
-      # If you are using :transaction, it will just rollback any additions
-      # when `.clean` is called. Any previous changes will be kept prior to
-      # the call `DatabaseCleaner.start`
-      #
-      # If you are using :truncation, it will erase everything once `.clean`
-      # is called.
-      seed_base_items_for_tests
-      # Always create organization, almost no tests can be run without one
-      create_organization
-
-      seed_with_default_records unless example.metadata[:skip_seed]
-    end
+  # Preparatifyication
+  config.before(:suite) do
+    DatabaseCleaner.clean_with(:truncation, except: %w[ar_internal_metadata])
+
+    # Stub out the Geocoder since we
+    Geocoder.configure(lookup: :test)
+
+    ["1500 Remount Road, Front Royal, VA 22630",
+     "123 Donation Site Way",
+     "Smithsonian Conservation Center new"].each do |address|
+       Geocoder::Lookup::Test.add_stub(
+         address, [
+           {
+             "latitude" => 40.7143528,
+             "longitude" => -74.0059731,
+             "address" => "1500 Remount Road, Front Royal, VA",
+             "state" => "Virginia",
+             "state_code" => "VA",
+             "country" => "United States",
+             "country_code" => "US"
+           }
+         ]
+       )
+     end
+
+    seed_base_data_for_tests
   end
 
-  config.before(:each, type: proc { |v| %i[request system controller].include?(v) }) do
-    create_default_users
-  end
+  config.before(:each, type: :system) do
+    unless File.exist?(Rails.root.join("public", "packs-task"))
+      Rails.logger.info "Detected missing webpack assets for testing. Running compilation step"
+      `NODE_ENV=test bin/webpack`
```

Comment:
One issue with this is that once JS has been compiled once, it'll never be recompiled. If you're making changes, you'd need to manually recompile it.

This is why I believe the previous attempt used a global variable to determine when the precompile step was run, and ran it just once when necessary. I think that approach probably makes more sense than this - basically we always re-run it when we run a system test, but only run it once per suite (as controlled by the global variable that only gets set once).

Reply:
Ahh that makes sense. However, I'am wondering what the best practices are? I don't think I've ever had to actually specify build webpack in tests.

Reply:
@dorner since this PR didn't add this but just moved it could we address this issue in another PR?

Reply:
Isn't this the same (or part of) the `assets:precompile` you're using above?

Reply:
Its the same, but I've actually figured out how to avoid having to do any of this.